### ISSUE TITLE
snowflake_tags (Data Source)

### ISSUE THREAD

**Issue by tsanton** [Snowflake-Labs/terraform-provider-snowflake#1372]
Hi!

I believe it would be useful with a `snowflake_tags` data source.
The usecase I'm facing is that we're declaring and managing tags in one project, but we'd like to reuse them for tag associations in another project.

Thinking it might look something like this .tf-side

```
data "snowflake_tags" "this" {
  database = "MY_DB"
  schema   = "TAG_SCHEMA"
}
```

.. and I guess a decent output will be the result of `show tags in schema my_db.tag_schema`:

```
database = MY_DB
schema = TAG_SCHEMA
tags = {
	tag_name_1: {
	        name  = tag_name_1
	        owner = someowner1
	        comment = foo
	        allowed_values = [""]
	},
	tag_name_2: {
                name  = tag_name_2
	        owner = someowner2
	        comment = bar
	        allowed_values = ["baz"]
	}
}
```

**Comment by sfc-gh-jmichalak** [comment]
Hi @tsanton   👋 

We've just released v0.99.0 ([release](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.99.0), [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#v0980--v0990)) in which we introduced `tags` data source. Please upgrade to this version and let us know if you have any issues.

**Comment by sfc-gh-jmichalak** [comment]
Closing due to inactivity.
